nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/service/NucleusTicketService.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.api.service;
+
+import io.github.nucleuspowered.nucleus.api.nucleusdata.Ticket;
+import org.apache.commons.lang3.tuple.Triple;
+import org.spongepowered.api.entity.living.player.User;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public interface NucleusTicketService {
+
+    /**
+     * Gets the ticket with the specified ID.
+     *
+     * @param id The ID of the ticket to get.
+     * @return The {@link Ticket} if present.
+     */
+    CompletableFuture<Optional<Ticket>> getTicketWithID(int id);
+
+    /**
+     * Gets all tickets (open and closed) owned by a {@link User}.
+     *
+     * @param user The {@link User} that has created the tickets.
+     * @return The {@link Ticket}s.
+     */
+    CompletableFuture<List<Ticket>> getTicketsForOwner(User user);
+
+    /**
+     * Gets all tickets (open and closed) assigned to a {@link User}.
+     *
+     * @param user The {@link User} assigned to the tickets.
+     * @return The {@link Ticket}s.
+     */
+    CompletableFuture<List<Ticket>> getTicketsForAssignee(User user);
+
+    /**
+     * Gets all tickets by their status, if they are open or closed.
+     *
+     * @param closed The status of the tickets to get, <code>true</code> if closed.
+     * @return The {@link Ticket}s.
+     */
+    CompletableFuture<List<Ticket>> getTicketsByStatus(boolean closed);
+
+    /**
+     * Gets all tickets with the specified argument, each Tri represents an argument, the first should be the column
+     * identifier, the second should be the matching operator (=, >, <, >=, <=, etc.), SQL matches such as BETWEEN can
+     * also be used., the third should be the value to match.
+     *
+     * @param arguments The arguments to fetch tickets
+     * @return The {@link Ticket}s which matched the query.
+     */
+    CompletableFuture<List<Ticket>> getTicketsByArguments(List<Triple<String, String, String>> arguments);
```

Comment:
Couple of thoughts:

* The `List` should possibly be a `Collection` or `Iterable`, whatever makes the most sense (not looked at the implementation yet, but just considering from a pure API view right now!)
* The column identfier is most _definitely_ an implementation detail. This should most likely be an enum that gets matched up in the implementation.
* The matching string is an implementation detail. Spin this out into an Enum or a set of keys, perhaps? This should reduce errors.
* Maybe consider making a subclass that contains these filters. Some operations don't make sense for some value types.

This might be better as a series of `TicketFilters` that can validate themselves so we guard against anything that doesn't make sense.